src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -503,6 +507,14 @@ private void runFafReplayFile(Path path) throws IOException {
       mapName = parseMapFolderName(rawReplayBytes);
     }
 
+    // For map generator games the map name is "None" because replay server gets map name by from DB based on filename
+    // from replay data, and DB does not contain generated maps.
+    if (StringUtils.equalsIgnoreCase(mapName, "None")) {
```

Comment:
This is also the case for other featured mods like coop.  How would that work out there?

Reply:
Coop maps seem to be handled in the lines directly above this code. Maybe it was null or empty in the past and due to a replay server change is "None" now for coop maps as well?